sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregate.scala

```diff
@@ -389,14 +379,16 @@ case class TungstenAggregate(
        UnsafeRow $keyTerm = (UnsafeRow) $iterTerm.getKey();
        UnsafeRow $bufferTerm = (UnsafeRow) $iterTerm.getValue();
        $outputCode
+
+       if (shouldStop()) return;
      }
 
-     $thisPlan.updatePeakMemory($hashMapTerm);
+     incPeakExecutionMemory($hashMapTerm.getPeakMemoryUsedBytes());
```

Comment:
can we bake the peak memory usage into hashMapTerm.free()? This seems like something we'll forget to do.